serve/graph/model/transaction.go

```diff
@@ -149,6 +149,22 @@ func (tr *TransactionResponse) Data() string {
 	return string(tr.response.Data)
 }
 
+func (tr *TransactionResponse) Events() []TransactionEvent {
+	events := make([]TransactionEvent, 0)
+
+	data, err := json.Marshal(tr.response.Events)
+	if err != nil {
+		return events
+	}
+
+	err = json.Unmarshal(data, &events)
+	if err != nil {
+		return events
+	}
+
+	return events
+}
```

Comment:
This is a bit strange, maybe:

```suggestion
func (tr *TransactionResponse) Events() []TransactionEvent {
	events := make([]TransactionEvent, 0)
	for _, e := tr.response.Events {
		events = append(events, NewTransactionEvent(e))
	}

	return events
}
```

If we can do the same we did with transaction messages to only parse the messages once, much better.

Reply:
Thank you for your suggestion.

I'm going to create an object per event, and change it to behave lazily.

Reply:
[6df815b](https://github.com/gnolang/tx-indexer/pull/43/commits/6df815bbbf83df643a69c471749b9246dd4d027c)